Lib/test/test_capi/test_opt.py

```diff
@@ -2305,6 +2305,37 @@ def testfunc(n):
         self.assertNotIn("_GUARD_TOS_INT", uops)
         self.assertNotIn("_GUARD_NOS_INT", uops)
 
+    def test_attr_promotion_failure(self):
+        # We're not testing for any specific uops here, just
+        # testing it doesn't crash.
+        result = script_helper.run_python_until_end('-c', textwrap.dedent("""
+        import _testinternalcapi
+        import _opcode
+        import email
+
+        def get_first_executor(func):
+            code = func.__code__
+            co_code = code.co_code
+            for i in range(0, len(co_code), 2):
+                try:
+                    return _opcode.get_executor(code, i)
+                except ValueError:
+                    pass
+            return None
+
+        def testfunc(n):
+            for _ in range(n):
+                email.jit_testing = None
+                prompt = email.jit_testing
+                del email.jit_testing
+
+
+        testfunc(_testinternalcapi.TIER2_THRESHOLD)
+        ex = get_first_executor(testfunc)
+        assert ex is not None
+        """), PYTHON_JIT="1")
```

Comment:
Is it necessary to pass `PYTHON_JIT` here? I would think that since this test requires the JIT anyway (via `requires_jit_enabled`), passing the variable would be redundant. Though another test passes it as well (added here: https://github.com/python/cpython/pull/129194/files#diff-67b5bd915e37f77df79907865982648172c27d3b96fbb072b30a73c6a2735439R649)